integration-libs/opf/base/core/services/opf-cart-handler.service.ts

```diff
@@ -165,7 +172,18 @@ export class OpfCartHandlerService {
       userId: this.cartHandlerState.userId,
       extraData: { active: true },
     });
-    return this.checkStableCart().pipe(
+    return this.activeCartFacade.getActiveCartId().pipe(
+      filter((cartId) => cartId === this.cartHandlerState.previousCartId),
```

Comment:
make sure original cart is back as active cart
then we check if stale cart (ex-active cart which will be deleted) and active cart are stable.